Mithril update 2025-07-02 (#617)
IntersectMBO/cardano-updates: blog/2025-07-02-mithril.md

---
title: Mithril Team Update
slug: 2025-07-02-mithril
authors: jpraynaud
tags: [mithril]
hide_table_of_contents: false
---

## High level overview

> This week, the Mithril team completed the implementation of the publisher and consumer for the DMQ network using the `n2c local message submission` and `n2c local message notification` mini-protocols from the Pallas library, and they worked on implementing a fake DMQ node for end to end testing. The team also completed the stabilization of the Cardano database v2 backend within the Mithril client CLI and library. Additionally, they kept implementing the certificate chain synchronization in the follower aggregator, prepared the upgrade to the Cardano node v.10.5, and enhanced some workflows in the CI.

Reviewer: ```suggestion
This week, the Mithril team completed the implementation of the publisher and consumer for the DMQ network using the `n2c local message submission` and `n2c local message notification` mini-protocols from the Pallas library.They also worked on creating a fake DMQ node to facilitate end to end testing.  The team also completed the stabilization of the Cardano database v2 backend within the Mithril client CLI and library. Additionally, they kept implementing the certificate chain synchronization in the follower aggregator, prepared the upgrade to the Cardano node v.10.5, and enhanced some workflows in the CI.
```

> Finally, they kept preparing the upgrade of the repository to Rust 2024 edition and continued refactoring the STM cryptographic library.

Reviewer: ```suggestion
Finally, they continued preparing the repository for the Rust 2024 edition upgrade and progressed with refactoring the STM cryptographic library.
```
